src/accessories/dzk-bacnet.ts

```diff
@@ -0,0 +1,991 @@
+      type: bacnet.enum.ApplicationTags.BACNET_APPLICATION_TAG_UNSIGNED_INT,
+      value: newState,
+    });
+    DzkZone.dzk_opmode = newState;
+    return DzkZone.dzk_opmode;
+  }
+  getCurrentTemperature(): Promise<number> {
+    return this.getZonePv("room-temperature");
+  }
+  getCurrentFanState(): Promise<number> {
+    /* controls for fan...
+     *
+     * global.dzk-global-fan
+     * global.iu-speed
+     * zone.local-ventilation
+     */
+    return DzkZone.getGlobalPv("dzk-global-fan");
+  }
+  getCurrentRelativeHumidity(): Promise<number> {
+    return this.getZonePv("humidity");
+  }
+  getHeatSetpoint(): Promise<number> {
+    return this.getZonePv("heat-set-point");
+  }
+  setHeatSetpoint(newTemp: number): Promise<number> {
+    return this.setZonePv("heat-set-point", newTemp);
+  }
+  getCoolSetpoint(): Promise<number> {
+    return this.getZonePv("cold-set-point");
+  }
+  setCoolSetpoint(newTemp: number): Promise<number> {
+    return this.setZonePv("cold-set-point", newTemp);
+  }
+
+  /*
+   * XXX-ELH: Obsolete. This was usable before I realized
+   *          that HAP provides the two above charaacteristics
+   *          for separate heat/cool setpoints (thesholds).
+   *
+   * setTargetTemperature is tricky because
+   * DZK has separate setpoints for heating and cooling
+   * but HAP only has the one "target temperature".
+   * There is no DZK object that shows whether we are in auto/heating mode or
+   * auto/cooling, so we can't really tell what set point to care about.
+   *
+   * As a little hack, we remember the last nonzero heating/cooling demand and
+   * use that to determine what setpoint to use.
+   *
+   * lastDemand is postive for heating demand, negative for cooling demand,
+   * zero for uninitialized.
+   */
+  private lastDemand = 0;
+
+  getTargetTemperature(): Promise<number> {
+    let bac_ob = "";
+    this.log.info(
+      "DzkZone.getTargetTemperature zone" + this.zno + ".lastDemand: ",
+      this.lastDemand
+    );
+    switch (DzkZone.dzk_opmode) {
+      case DzkOperationMode.AUTO:
+      case DzkOperationMode.DRY:
+        bac_ob = this.lastDemand > 0 ? "heat-set-point" : "cold-set-point";
+        break;
+      case DzkOperationMode.COOL:
+        bac_ob = "cold-set-point";
+        break;
+      case DzkOperationMode.HEAT:
+        bac_ob = "heat-set-point";
+        break;
+    }
+    return this.getZonePv(bac_ob);
+  }
+  setTargetTemperature(newTemp: number): Promise<number> {
+    let bac_ob = "";
+    this.log.info(
+      "DzkZone.setTargetTemperature zone" + this.zno + ".lastDemand: ",
+      this.lastDemand
+    );
+    switch (DzkZone.dzk_opmode) {
+      case DzkOperationMode.AUTO:
+      case DzkOperationMode.DRY:
+        bac_ob = this.lastDemand > 0 ? "heat-set-point" : "cold-set-point";
+        break;
+      case DzkOperationMode.COOL:
+        bac_ob = "cold-set-point";
+        break;
+      case DzkOperationMode.HEAT:
+        bac_ob = "heat-set-point";
```

Comment:
The switch statement seems to not be used. When returning, the calculations made in the switch statement aren't used. Is it meant to be this way?